app/models/project_services/composer_service.rb

```diff
@@ -0,0 +1,464 @@
+              ['Tusk Asset', 'tusk-asset'],
+              ['Tusk Command', 'tusk-command'],
+              ['Tusk Task', 'tusk-task'],
+              ['TYPO3 CMS Extension', 'typo3-cms-extension'],
+              ['TYPO3 Flow Boilerplate', 'typo3-flow-boilerplate'],
+              ['TYPO3 Flow Build', 'typo3-flow-build'],
+              ['TYPO3 Flow Framework', 'typo3-flow-framework'],
+              ['TYPO3 Flow Package', 'typo3-flow-package'],
+              ['TYPO3 Flow Plugin', 'typo3-flow-plugin'],
+              ['TYPO3 Flow Site', 'typo3-flow-site'],
+              ['Wolf CMS Plugin', 'wolfcms-plugin'],
+              ['WordPress Core', 'wordpress-core'],
+              ['WordPress Must Use Plugin', 'wordpress-muplugin'],
+              ['WordPress Plugin', 'wordpress-plugin'],
+              ['WordPress Theme', 'wordpress-theme'],
+              ['Zend Extra', 'zend-extra'],
+              ['Zend Library', 'zend-library'],
+              ['Zend Module', 'zend-module'],
+              ['Zikula Module', 'zikula-module'],
+              ['Zikula Theme', 'zikula-theme']
+            ],
+            default_choice: 'library'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Branches:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_branches'
+          },
+          { type: 'text',
+            name: 'branch_filters',
+            placeholder: 'branches you wish to export comma separated.',
+            hint: 'Separate branches with commas. '\
+                  'Leave blank to export all branches.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Tags:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_tags'
+          },
+          { type: 'text',
+            name: 'tag_filters',
+            placeholder: 'tags you wish to export comma separated.',
+            hint: 'Separate tags with commas. '\
+                  'Leave blank to export all tags.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Advanced:', fields:
+        [
+          { type: 'textarea',
+            name: 'custom_json',
+            placeholder: 'custom json to use for exporting this package.'
+          }
+        ]
+      }
+    ]
+  end
+
+  # disable test button
+  def can_test?
+    false
+  end
+
+  def supported_events
+    %w(push tag_push)
+  end
+
+  def execute(push_data)
+    process_project
+  end
+
+  def process_project
+
+    # delete the project repository file since it will be regenerated
+    if File.exists?(repo_path)
+      File.delete(repo_path)
+    end
+
+    if activated?
+      # process packages for all tags
+      project.repository.tags.each do |tag|
+        process_commit(tag)
+      end
+
+      # process packages for all branches
+      project.repository.branches.each do |branch|
+        process_commit(branch)
+      end
+
+      repository.write
+    end
+    update_root_file
+  rescue
+    # skip on error
```

Comment:
What about log? What kind of errors can occur here? Maybe we want to log those somewhere? At this point we are catching all errors, I think it would be good to narrow it down a bit :-)